source/vibe/web/rest.d

```diff
@@ -1,10 +1,226 @@
-/**
-	Automatic REST interface and client code generation facilities.
+// TODO: sub-interfaces
 
-	Copyright: © 2012-2013 RejectedSoftware e.K.
-	License: Subject to the terms of the MIT license, as written in the included LICENSE.txt file.
-	Authors: Sönke Ludwig, Михаил Страшун
-*/
+/**
+ * Module to work with REST API, from both server and client side.
+ *
+ * This modules aims to provide a typesafe way to deal with REST API.
+ * As they are quite a lot of similarities between them, this module use
+ * D interfaces as the way to represent REST API.
+ * While REST is not HTTP specific, the vast majority of use cases rely on it,
+ * and so does this module.
+ *
+ * This module, while being usable and more convenient from both client and
+ * server side, is also usable solely for its client or server implementation.
+ * The following documentation will explain in details how this module work,
+ * without any specifics about the client or server side.
+ * Since each implementation only require the use of one symbol from here,
+ * specifics are explained on those. You can thus find server only
+ * documentation on $(D registerRestInterface) and client only documentation
+ * on $(D RestInterfaceClient).
+ *
+ * There are few advantages in using this module over the more basic `vibe.http`
+ * package provided:
+ * - Automatic client generation: once the interface is defined, it will be used
+ *   both by the client side and the server side, which means there is no way
+ *   to have a mismatch between your client and server code.
+ * - Automatic route generation for the server: one of the job of the
+ *   REST module is to generate the route for your API.
+ * - Automatic serialization / deserialization: Instead of doing your own
+ *   serialization and deserialization, you just design normal member functions
+ *   and let us take care of the heavy lifting. You still have the possibility
+ *   to send direct JSON input by defining a parameter as Json, or pure string
+ *   by returning a string.
+ * - Higher level representation integrated into D: Some concepts of the
+ *   interfaces, such as optional parameter or header return values feat
+ *   nicely in D by the use of `std.typecons.Nullable` and `out` parameters.
+ *
+ *
+ * The most basic interface that can be defined is as follow:
+ * ----
+ * @path("/api/")
+ * interface APIRoot {
+ *     string get();
+ * }
```

Comment:
I didn't document `@rootPath` for obvious reasons.
Likely, I didn't document `@rootPathFromName` yet. I don't expect this to be pulled before 0.7.24, and by then, as you might remember, I'd like to revisit the default behaviour.